Sources/Quick/Hooks/ExampleHooks.swift

```diff
@@ -38,7 +38,37 @@ final internal class ExampleHooks {
     }
 
     internal func appendAround(_ closure: @escaping AroundExampleClosure) {
-        wrappers.append { _, runExample in closure(runExample) }
+        wrappers.append { _, runExample in await closure(runExample) }
+    }
+
+    /// Synchronous version of aroundEach, passing in metadata to the closure.
+    /// Warning: This should only be used for objective-c compatibility.
+    internal func appendAroundSync(_ closure: @escaping AroundExampleWithMetadataSyncClosure) {
+        wrappers.append { exampleMetadata, runExample in
+            closure(exampleMetadata) {
```

Comment:
I struggled with this for a while, until I settled on this hack.

This is a code workaround for the fact that the example can be async, but because this is used from objc, and our objective-c API does not actually allow async things, it's actually sync (ish, it still hands off the thread while the `runExample` block runs).